scene/theme/theme_db.cpp

```diff
@@ -198,21 +198,21 @@ Ref<StyleBox> ThemeDB::get_fallback_stylebox() {
 	return fallback_stylebox;
 }
 
-void ThemeDB::get_native_type_dependencies(const StringName &p_base_type, List<StringName> *p_list) {
-	ERR_FAIL_NULL(p_list);
+void ThemeDB::get_native_type_dependencies(const StringName &p_base_type, LocalVector<StringName> &r_result) {
+	if (p_base_type == StringName()) {
+		return;
+	}
 
 	// TODO: It may make sense to stop at Control/Window, because their parent classes cannot be used in
 	// a meaningful way.
-	StringName class_name = p_base_type;
-	while (class_name != StringName()) {
-		p_list->push_back(class_name);
-		class_name = ClassDB::get_parent_class_nocheck(class_name);
-	}
+	r_result.reserve(8);
```

Comment:
How was this `8` chosen? An arbitrary number may spell disaster...

Reply:
It's just reserving, it doesn't change anything except worsening performance if the guess is wrong

Reply:
In that case wouldn't it be better to reserve memory in `ClassDB::get_parent_class_nocheck` itself? I know it sounds odd but ClassDB would "know" better (if someone decided to expand the code). Makes the intent more clear, too.

Reply:
8's just an estimate of how many classes might be added, ideally a little higher than needed but it should be fine if it needs to grow past that in some cases.  It's mainly to avoid a bunch of small allocations for 1/2/4 items in the vector.  It's based on how many are expected to be added by `ClassDB`, so agreed on moving it there.

Ideally, it would still be reserved before adding `p_base_type`, which might not be in `ClassDB`, so could be unexpected to always add it in there.  Might shuffle things around a bit to make `ClassDB::get_parent_classes_nocheck` match up better with `ClassDB::is_parent_class` (include the passed in class if it's found), and have `ThemeDB::get_native_type_dependencies` add `p_base_type` when needed (if `ClassDB` doesn't find it).

Reply:
Made the changes above and renamed `get_parent_classes_nocheck` to `get_inheritance_chain_nocheck`, the old name felt confusing since the starting class is now included.